src/Uno.Extensions.Reactive/Core/Axes/SelectionIndexRange.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Linq;
+
+namespace Uno.Extensions.Reactive;
+
+/// <summary>
+/// An index based range of selected items.
+/// </summary>
+/// <param name="FirstIndex">The index of the first selected item.</param>
+/// <param name="Length">The count of selected items.</param>
+public sealed record SelectionIndexRange(uint FirstIndex, uint Length)
+{
+	/// <summary>
+	/// Gets the index of the last selected item.
+	/// </summary>
+	public uint LastIndex { get; } = FirstIndex + Length - 1;
+
+	/// <inheritdoc />
+	public override string ToString()
+		=> $"[{FirstIndex}, {LastIndex}[";
+};
```

Comment:
Why aren't you using the C# 8's `System.Range` stuff?

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/ranges#systemrange

Reply:
`System.Range` is only starting from netstd 2.1 while we still target 2.0. I tried to mimic the contract of `ItemdIndexRange` https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.data.itemindexrange?view=winrt-22621 which is part of the selection contract of WinUI, but indeed it could interesting to get closer to the `System.Range`. Will investigate advantages of both in future dev, will however merge as-is as this PR is a blocker for other work.